GramAddict/core/interaction.py

```diff
@@ -164,61 +208,132 @@ def _on_interaction(
             )
-            return False
-        elif followback_button.exists():
-            logger.info(
-                f"@{username} already follows you.", extra={"color": f"{Fore.GREEN}"}
+            followback_button = device.find(
+                classNameMatches=BUTTON_REGEX,
+                clickable=True,
+                textMatches=FOLLOWBACK_REGEX,
             )
+            if unfollow_button.exists():
+                logger.info(
+                    f"You already follow @{username}.", extra={"color": f"{Fore.GREEN}"}
+                )
+                return False
+            elif followback_button.exists():
+                logger.info(
+                    f"@{username} already follows you.",
+                    extra={"color": f"{Fore.GREEN}"},
+                )
+                return False
+            else:
+                logger.error(
+                    "Cannot find neither Follow button, Follow Back button, nor Unfollow button. Maybe not English language is set?"
+                )
+                save_crash(device)
+                switch_to_english(device)
+                raise LanguageNotEnglishException()
+
+        follow_button.click()
+        detect_block(device)
+        logger.info(f"Followed @{username}", extra={"color": f"{Fore.GREEN}"})
+        random_sleep()
+        return True
+    else:
+        logger.info("Reached total follows limit, not following.")
+        return False
+
+
+def _on_watch(sessions, session_state):
+    session_state = sessions[-1]
+    session_state.totalWatched += 1
+
+
+def _watch_stories(
+    device,
+    profile_view,
+    username,
+    stories_to_watch,
+    stories_percentage,
+    on_watch,
+    args,
+    session_state,
+):
+    if not session_state.check_limit(args, limit_type="WATCH", output=False):
+        story_chance = randint(1, 100)
+        if story_chance > stories_percentage:
             return False
-        else:
-            logger.error(
-                "Cannot find neither Follow button, Follow Back button, nor Unfollow button. Maybe not English language is set?"
-            )
-            save_crash(device)
-            switch_to_english(device)
-            raise LanguageNotEnglishException()
 
-    follow_button.click()
-    detect_block(device)
-    logger.info(f"Followed @{username}", extra={"color": f"{Fore.GREEN}"})
-    random_sleep()
-    return True
+        stories_to_watch = get_value(stories_to_watch, "Stories count: {}", 0)
+
+        if stories_to_watch > 6:
+            logger.error("Max number of stories per user is 6")
+            stories_to_watch = 6
+
+        if stories_to_watch == 0:
+            return False
+
+        if profile_view.isStoryAvailable():
+            profile_picture = profile_view.profileImage()
+            if profile_picture.exists():
+                profile_picture.click()  # Open the first story
+                on_watch()
+                sleep(uniform(1.5, 2.5))
```

Comment:
please, use `random_sleep` method. If there is a need to change the default limits, please customize it by passing inferior and superior limits.